### PR TITLE
[DOC] JSON command edits

### PR DESCRIPTION
**Pull request by nermiller**
Updates JSON commands and restructures sections

### CODE REVIEW

File: docs/commands/json.arrlen.md [hunk 1]
```diff
@@ -1,25 +1,72 @@
...
+## Optional arguments
+
+<details open><summary><code>path</code></summary> 
+
+is JSONPath to specify. Default is root `$`, if not provided. Returns null if the `key` or `path` do not exist.
+</details>
+
+## Return
+
+JSON.ARRLEN returns by recursive descent an array of integer replies for each path, the array's length, or `nil`, if the matching JSON value is not an array.
```
Comment (oshadmi) [inline docs/commands/json.arrlen.md:21]: Not necessarily recursive descent. Seems rephrasing is needed, such as:
```suggestion
`JSON.ARRLEN` returns an [array](/docs/reference/protocol-spec/#resp-arrays) of integer replies, an integer for each matching value, each is the array's length, or `nil`, if the matching value is not an array.
```

File: docs/commands/json.resp.md [hunk 2]
```diff
@@ -1,14 +1,82 @@
...
+<details open><summary><code>path</code></summary> 
+
+is JSONPath to specify. Default is root `$`. This command uses the following mapping from JSON to RESP:
+
+*   JSON `null` maps to the bulk string reply.
+*   JSON `false` and `true` values map to the simple string reply.
+*   JSON number maps to the integer reply or bulk string reply, depending on type.
+*   JSON string maps to the bulk string reply.
+*   JSON array is represented as an array reply in which the first element is the simple string reply `[`, followed by the array's elements.
+*   JSON object is represented as an aarray reply in which the first element is the simple string reply `{`. Each successive entry represents a key-value pair as a two-entry array reply of the bulk string reply.
```
Comment (oshadmi) [inline docs/commands/json.resp.md:23]:
```suggestion
*   JSON object is represented as an array reply in which the first element is the simple string reply `{`. Each successive entry represents a key-value pair as a two-entry array reply of the bulk string reply.
```

File: docs/commands/json.set.md [hunk 3]
```diff
@@ -1,48 +1,89 @@
...
-@return
+is key to modify.
+</details>
 
-@simple-string-reply - `OK` if executed correctly, or @nil-reply if the specified `NX` or `XX`
-conditions were not met.
+<details open><summary><code>value</code></summary> 
 
-@examples
+is one or more values to index in one or more arrays.
```
Comment (oshadmi) [inline docs/commands/json.set.md:14]: `JSON.SET` accepts a single value to set
```suggestion
is value to set at the specified path
```

File: docs/commands/json.strappend.md [hunk 4]
```diff
@@ -1,20 +1,54 @@
...
 
-@examples
+is key to modify.
+</details>
 
-```
-redis> JSON.SET doc $ '{"a":"foo", "nested": {"a": "hello"}, "nested2": {"a": 31}}'
+<details open><summary><code>value</code></summary> 
+
+is one or more values to append to one or more strings.
```
Comment (oshadmi) [inline docs/commands/json.strappend.md:14]: `JSON.STRAPPEND` accepts a single value (unlike `JSON.ARRAPPEND`)
```suggestion
is value to append to one or more strings. 
```

File: docs/commands/json.toggle.md [hunk 5]
```diff
@@ -1,20 +1,74 @@
...
 
-@array-reply of @integer-reply - for each path, the new value (0-false or 1-true), or @nil-reply element for JSON values matching the path which are not boolean.
+## Required arguments
 
-@examples
+<details open><summary><code>key</code></summary> 
 
-```
-redis> JSON.SET doc $ '{"bool": true}'
+is key to parse.
```
Comment (oshadmi) [inline docs/commands/json.toggle.md:9]: Key is not being parsed
```suggestion
is key to modify.
```

Also relevant for other occurrences.
For read-commands, such as `JSON.TYPE` or `JSON.ARRLEN`, etc.
which do not modify the key, we can use
```
is key to read.
```

**Review by oshadmi — COMMENTED** [review]
A few more comments 🙏🏼

File: docs/commands/json.arrinsert.md [hunk 6]
```diff
@@ -1,28 +1,93 @@
...
 
-@examples
+is key to modify.
+</details>
 
-```
-redis> JSON.SET doc $ '{"a":[3], "nested": {"a": [3,4]}}'
+<details open><summary><code>value</code></summary> 
+
+is one or more values to append to one or more arrays.
```
[the remark this answers is not in the document]
Comment (nermiller) [reply docs/commands/json.arrinsert.md:14]: it should be insert, yes. copy pasta! i'll fix it

Comment (nermiller) [inline docs/commands/json.arrinsert.md:14]:
```suggestion
is one or more values to insert in one or more arrays. 
```

File: docs/commands/json.arrindex.md [hunk 7]
```diff
@@ -1,29 +1,107 @@
...
+{{< / highlight >}}
+
+Get the updated list of colors.
+
+{{< highlight bash >}}
+127.0.0.1:6379> JSON.GET item:1 $.colors
+"[[\"black\",\"silver\",\"yellow\",\"gold\",\"blue\"]]"
+{{< / highlight >}}
+
+Find the place where color silver is located.
```
Comment (nermiller) [inline docs/commands/json.arrindex.md:91]:
```suggestion
Find the place where color `silver` is located.
```

File: docs/commands/json.arrtrim.md [hunk 8]
```diff
@@ -1,35 +1,95 @@
...
-2) (nil)
-redis> JSON.GET doc $
-"[{\"a\":[2],\"nested\":{\"a\":false}}]"
-```
+{{< / highlight >}}
+
+Add new sound level values to the second product.
+
+{{< highlight bash >}}
+127.0.0.1:6379> JSON.ARRAPPEND key $.[1].max_level 200 300 400 500 600 700 800 1000
```
[the remark this answers is not in the document]
Comment (nermiller) [reply docs/commands/json.arrtrim.md:61]: yeah, i'll update that lol

File: docs/commands/json.arrpop.md [hunk 9]
```diff
@@ -1,28 +1,84 @@
...
+{{< / highlight >}}
+
+Get the updated array.
+
+{{< highlight bash >}}
+127.0.0.1:6379> JSON.GET key $.[1].max_level
+"[[90,100,120]]"
+{{< / highlight >}}
+
+Now insert a new lowest value in the second product.
```
[the remark this answers is not in the document]
Comment (nermiller) [reply docs/commands/json.arrpop.md]: fixed!

File: docs/commands/json.arrindex.md [hunk 10]
```diff
@@ -1,29 +1,107 @@
...
```
[the remark this answers is not in the document]
Comment (nermiller) [reply docs/commands/json.arrindex.md]: fixed! will rereview to ensure we treat them like code.

File: docs/commands/json.arrpop.md [hunk 11]
```diff
@@ -1,28 +1,84 @@
...
+{{< / highlight >}}
+
+Get all maximum values for the second product.
+
+{{< highlight bash >}}
+127.0.0.1:6379> JSON.GET key $.[1].max_level
+"[[80,90,100,120]]"
+{{< / highlight >}}
+
+Pop the lowest value because it's a duplicate of the highest value in the first product's index.
```
[the remark this answers is not in the document]
Comment (nermiller) [reply docs/commands/json.arrpop.md]: done!

File: docs/commands/json.arrinsert.md [hunk 12]
```diff
@@ -1,28 +1,93 @@
-Inserts the `json` values into the array at `path` before the `index` (shifts to the right).
+Insert the `json` values into the array at `path` before the `index` (shifts to the right)
```
[the remark this answers is not in the document]
Comment (nermiller) [reply docs/commands/json.arrinsert.md:1]: is that not expected?

File: docs/commands/json.arrinsert.md [hunk 13]
```diff
@@ -1,28 +1,93 @@
...
+## Optional arguments
+
+<details open><summary><code>path</code></summary> 
+
+is JSONPath to specify. Default is root `$`.
+</details>
+
+## Return value 
+
+JSON.INSERT returns an array of integer replies for each path, the array's new size, or `nil`, if the matching JSON value is not an array.
```
[the remark this answers is not in the document]
Comment (oshadmi) [reply docs/commands/json.arrinsert.md:35]: It makes sense to not add backticks when a link to the current doc would be generated,
However, backticks also affect formatting, allowing a common look for all of our commands in all pages.
(and a minor consideration: if copied elsewhere, would still be a valid link)

Comment (nermiller) [inline docs/commands/json.arrinsert.md:35]:
```suggestion
`JSON.ARRINSERT` returns an array of integer replies for each path, the array's new size, or `nil`, if the matching JSON value is not an array. 
```

File: docs/commands/json.arrinsert.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (oshadmi) [reply docs/commands/json.arrinsert.md:1]: I assume most users would expect that, but as far as I am concerned, if we can be formal about indices being zero-based, why not?

[the remark this answers is not in the document]
Comment (nermiller) [reply docs/commands/json.arrinsert.md:1]: sure, will add.

**Comment by nermiller** [comment]
@oshadmi, I believe I addressed most if not all of your comments. If there are any minor comments left, let's note and pick them up as we go. Please take another look. Thank you for persevering through this!

File: docs/commands/json.arrappend.md [hunk 14]
```diff
@@ -42,7 +42,7 @@ Create a document for noise-cancelling headphones in black and silver colors.
 OK
 {{< / highlight >}}
 
-Add color blue to the end of the `colors` array. JSON.ARRAPEND returns the array's new size.
+Add color `"blue"` to the end of the `colors` array. `JSON.ARRAPEND` returns the array's new size.
```
Comment (oshadmi) [inline docs/commands/json.arrappend.md:45]: I noticed we do not quote values in the description of other examples
```suggestion
Add color `blue` to the end of the `colors` array. `JSON.ARRAPEND` returns the array's new size.
```

File: docs/commands/json.arrinsert.md [hunk 15]
```diff
@@ -32,7 +32,7 @@ is JSONPath to specify. Default is root `$`.
 
 ## Return value 
 
-JSON.INSERT returns an array of integer replies for each path, the array's new size, or `nil`, if the matching JSON value is not an array. 
+`JSON.ARRINSERT` returns an array of integer replies for each path, the array's new size, or `nil`, if the matching JSON value is not an array.
```
Comment (oshadmi) [inline docs/commands/json.arrinsert.md:35]:
```suggestion
`JSON.ARRINSERT` returns an [array](/docs/reference/protocol-spec/#resp-arrays) of integer replies for each path, the array's new size, or `nil`, if the matching JSON value is not an array. 
```

File: docs/commands/json.arrpop.md [hunk 16]
```diff
@@ -23,7 +23,7 @@ is JSONPath to specify. Default is root `$`.
 
 ## Return
 
-JSON.ARRPOP returns an array of bulk string replies for each path, the popped JSON value, or `nil`, if the matching JSON value is not an array.
+`JSON.ARRPOP` returns an array of bulk string replies for each path, each reply is the popped JSON value, or `nil`, if the matching JSON value is not an array.
```
Comment (oshadmi) [inline docs/commands/json.arrpop.md:26]:
```suggestion
`JSON.ARRPOP` returns an [array](/docs/reference/protocol-spec/#resp-arrays) of bulk string replies for each path, each reply is the popped JSON value, or `nil`, if the matching JSON value is not an array.
```

File: docs/commands/json.arrtrim.md [hunk 17]
```diff
@@ -58,18 +58,18 @@ OK
...
 Get the updated array.
 
 {{< highlight bash >}}
 127.0.0.1:6379> JSON.GET key $.[1].max_level
-"[[85,90,100,120,200,300,400,500,600,700,800,1000]]"
+"[[85,90,100,120,140,160,180,200,220,240,260,280]]"
 {{< / highlight >}}
 
-Trim all values between the fourth and the eight element, inclusive of that last element.
+Keep only the values between the fifth and the eight element, inclusive of that last element.
```
Comment (oshadmi) [inline docs/commands/json.arrtrim.md:72]: I missed this
```suggestion
Keep only the values between the fifth and the ninth element, inclusive of that last element.
```

**Review by oshadmi — COMMENTED** [review]
Few comments, last touch-up

**Review by oshadmi — APPROVED** [review]
👍🏼 👌🏼

**Pull request merged**